service/gcs/core/gcs/storage.go

```diff
@@ -104,32 +141,52 @@ func (c *gcsCore) deviceIDToName(deviceID string) (string, error) {
 	if len(deviceNames) > 1 {
 		return "", errors.Errorf("more than one block device could match SCSI ID \"%s\"", scsiID)
 	}
-	return deviceNames[0].Name(), nil
+	return "/dev/" + deviceNames[0].Name(), nil
+
+}
+
+// deviceIDToName converts a device ID (scsi:<lun> or pmem:<device#> to a
+// device name (/dev/sd? or /dev/pmem?).
+// For temporary compatibility, this also accepts just <lun> for SCSI devices.
+func deviceIDToName(id string) (device string, pmem bool, err error) {
+	const (
+		pmemPrefix = "pmem:"
+		scsiPrefix = "scsi:"
+	)
+
+	if strings.HasPrefix(id, pmemPrefix) {
+		return "/dev/pmem" + id[len(pmemPrefix):], true, nil
+	}
+
+	lunStr := id
+	if strings.HasPrefix(id, scsiPrefix) {
+		lunStr = id[len(scsiPrefix):]
+	}
+
+	if lun, err := strconv.ParseInt(lunStr, 10, 8); err == nil {
+		name, err := scsiLunToName(int(lun))
+		return name, false, err
+	}
+
+	return "", false, errors.Errorf("unknown device ID %s", id)
 }
 
 // mountMappedVirtualDisks mounts the given disks to the given directories,
 // with the given options. The device names of each disk are given in a
 // parallel slice.
-func (c *gcsCore) mountMappedVirtualDisks(disks []prot.MappedVirtualDisk, devices []string) error {
-	if len(disks) != len(devices) {
-		return errors.Errorf("disk and device slices were of different sizes. disks: %d, devices: %d", len(disks), len(devices))
+func (c *gcsCore) mountMappedVirtualDisks(disks []prot.MappedVirtualDisk, mounts []*mountSpec) error {
+	if len(disks) != len(mounts) {
+		return errors.Errorf("disk and device slices were of different sizes. disks: %d, mounts: %d", len(disks), len(mounts))
```

Comment:
It certainly is the uncommon path so maybe it doesn't matter but given you have already counted once. Just wondering if its better to:

```go
if i, j := len(disks), len(mounts); i != j {
// use I, J here.
}
```